File: test/benchmark/cases/generic_mix/case.java
class Main {
    static int mixInt(int value, int salt) {
        return (value * 17 + salt) % 1009;
    }

    static long mixLong(long value, long salt) {
        return (value * 17 + salt) % 1009;
    }

    public static void main(String[] args) {
        var raw = System.getenv("CP_BENCH_INPUT");
        if(raw == null) {
            System.exit(1);
        }

        var count = Integer.parseInt(raw);
        var index = 0;
        var totalI32 = 0;
        var totalI64 = 0L;
        while(index < count) {
            totalI32 = (totalI32 + mixInt(index, 11)) % 1000003;
            totalI64 = (totalI64 + mixLong(index, 23)) % 1000003;
            index += 1;
        }
        System.exit((int)((totalI32 + totalI64) % 251));
    }
}

File: test/benchmark/cases/generic_mix/case.cpp
#include <cstdint>
#include <cstdlib>

template<typename T>
auto mix(T value, T salt) -> T
{
    return (value * 17 + salt) % 1009;
}

auto main() -> int
{
    auto const* raw = std::getenv("CP_BENCH_INPUT");
    if(raw == nullptr) {
        return 1;
    }

    auto const count = std::atoi(raw);
    auto index = 0;
    auto total_i32 = 0;
    auto total_i64 = std::int64_t{ 0 };
    while(index < count) {
        total_i32 = (total_i32 + mix<int>(index, 11)) % 1000003;
        total_i64 = (total_i64 + mix<std::int64_t>(index, 23)) % 1000003;
        index += 1;
    }
    return static_cast<int>((total_i32 + total_i64) % 251);
}
